include/RAJA/exec-cuda/MemUtils_CUDA.hxx

```diff
@@ -63,12 +64,16 @@ namespace RAJA {
 
 #define RAJA_CUDA_REDUCE_BLOCK_LENGTH (1024 + 8) * 16
 
+
 ///
 /// Typedef defining common data type for RAJA-Cuda reduction data blocks
 /// (use this in all cases to avoid type confusion).
 ///
 typedef double CudaReductionBlockDataType;
 
+typedef struct{double val; Index_type idx;} CudaReductionLocBlockDataType; 
```

Comment:
To be honest, I am also concerned about CudaReductionLocBlockDataType.  I'm wondering if we can somehow use the template type, or at least use a union of scalar types for the 'val' member (realizing the the largest will determine the size of the data structure overall.